packages/playground/react-jsx-runtime/App.jsx

```diff
@@ -0,0 +1,54 @@
+/* eslint-disable */
+import { useState } from 'react'
+
+const StatelessCounter = ({ value, onChange }) => {
+  return (
+    <button onClick={() => onChange((count) => count + 1)}>
+      count is: {value}
+    </button>
+  )
+}
+
+const withState = (Comp) => {
+  return ({ defaultValue }) => {
+    const [val, setCount] = useState(defaultValue)
+    return <Comp value={val} onChange={setCount} />
+  }
+}
+
+const withWrapper = (Comp, Wrapper) => {
+  return (props) => {
+    return (
+      <Wrapper>
+        <Comp {...props} />
+      </Wrapper>
+    )
+  }
+}
+
+// @ts-expect-error ts does not support pipeline operators yet
+const Counter = StatelessCounter |> withState |> ((_) => withWrapper(_, 'p'))
```

Comment:
TypeScript is reporting `Expression expected. ts(1109)` on this line. Also, the optimizer is choking on this syntax when scanning for imports. We should use another plugin for now.

Reply:
Ops. I think this worked because of cache thus import analysis was skipped without choking. I will find one that is esbuild compatibility

Reply:
After some investigations, I think we cannot add parser/plugin for stage < 3 proposals in a vote playground, since stage < 3 proposal is not supported by both TS & esbuild, see a related issue https://github.com/vitejs/vite/issues/2349

Reply:
I updated the playground to demo how to enable `@emotion/react` with `@vite/plugin-react` to fix a wanted issue: https://github.com/vitejs/vite/issues/3276
Also added a test for it, which is similar to the original `playground/react`, without `file location metadata` check.
However, I am not sure what happened for this test, but the tests are now failing on windows.
https://github.com/pengx17/vite/runs/3150400557?check_suite_focus=true

It is reporting the following error when running on Windows/Node 12 for build test
```
    [vite]: Rollup failed to resolve import "D:aiteite
    ode_modules
    eactjsx-runtime.js" from "react/jsx-runtime".
    This is most likely unintended because it can break your application at runtime.
    If you do want to externalize this module explicitly add it to
    `build.rollupOptions.external`
```

Weird that this issue does not happen for other scenarios.

Reply:
Fixed with https://github.com/aleclarson/vite/pull/3/commits/e6bcfae5ed79b1ff74a9885230ab9df9c00dc263, I think

Reply:
Confirmed and the tests have passed